File: unity/Assets/Scripts/Multiplayer/Games/PlayerShort.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using UnityEngine;
using static Michitai.Multiplayer.Time.Time;

namespace Michitai.Multiplayer.Games
{
    /// <summary>
    /// Shortened player information for listing players in Unity.
    /// Contains essential player data with timestamp parsing support.
    /// </summary>
    [System.Serializable]
    public class PlayerShort
    {
        [SerializeField]
        private string last_login;
        [SerializeField]
        private string created_at;

        /// <summary>
        /// The unique ID of the player.
        /// </summary>
        public int id;

        /// <summary>
        /// The player's display name.
        /// </summary>
        public string player_name;

        /// <summary>
        /// Whether the player is currently online.
        /// </summary>
        public bool is_online;

        /// <summary>
        /// Timestamp of the player's last login (parsed from string).
        /// </summary>
        public DateTimeOffset? LastLogin
        {
            get
            {
                return ParseUtc(last_login);
            }
        }

        /// <summary>
        /// Timestamp when the player account was created (parsed from string).
        /// </summary>
        public DateTimeOffset? CreatedAt
        {
            get
            {
                return ParseUtc(created_at);
            }
        }
    }
}

File: java/src/main/java/com/michitai/multiplayer/games/PlayerShort.java
package com.michitai.multiplayer.games;

import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * Short player information for player listings.
 */
public class PlayerShort {
    @JsonProperty("id")
    private int id;

    @JsonProperty("player_name")
    private String playerName;

    @JsonProperty("is_online")
    private boolean isOnline;

    @JsonProperty("last_login")
    private String lastLogin;

    @JsonProperty("created_at")
    private String createdAt;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getPlayerName() {
        return playerName;
    }

    public void setPlayerName(String playerName) {
        this.playerName = playerName;
    }

    public boolean isOnline() {
        return isOnline;
    }

    public void setOnline(boolean online) {
        isOnline = online;
    }

    public String getLastLogin() {
        return lastLogin;
    }

    public void setLastLogin(String lastLogin) {
        this.lastLogin = lastLogin;
    }

    public String getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(String createdAt) {
        this.createdAt = createdAt;
    }
}
